DOC-3729: Added a gradle wrapper environment and created antora build script (#79)
DataStaxDocs/datastax-streaming-docs-site: README.adoc

| /learning
|===

> *assuming a URL prefix of https://docs.datastax.com/en/streaming

Reviewer: What does the asterisk at the beginning of the line do? I need education on that - looks like a mistake to me, but I figure it isn't.

git clone https://github.com/datastax/streaming-learning-docs
----

. Build the site to ensure the local playbook is working properly. Note this could take quite a while to run the first time (~20-30 mins). Subsequent builds will be much faster.
+
[source,shell]
----
cd datastax-streaming-docs-site
./gradlew -PtoDir="build/site" antora --playbook ./antora-playbooks/develop.yaml
----

> . If built correctly you should have a new `build/site folder that holds the compiled Streaming documentation site.

Reviewer: ```suggestion
. If built correctly, a new `build/site` folder that holds the compiled Streaming documentation site will be present.
```

== Developing new documentation

With the above folder structure in place you can now edit any of the documentation areas. There are a few key things to keep in mind.

. Each documentation area is a separate git repo. You can edit the files in your local folder, but you'll need to commit and push those changes to the repo.

> . Building the streaming docs site means to use a playbook to clone and build each documentation area. Each playbook utilizes different git schemes for retrieving the src.

Reviewer: ```suggestion
. Building the streaming docs site means using a playbook to clone and build each documentation area. Each playbook utilizes different git schemes for retrieving the docs source.
```

> . Typically, for local desktop development you will use the "develop" playbook. Unlike the other playbook, this uses the branch of each docs area that oyu have checked out (HEAD). This means you can make changes to the docs and see them reflected in the streaming docs site without having to commit and push.

Reviewer: ```suggestion
. Typically, for local desktop development, you will use the "develop" playbook. Unlike the other playbook, the "develop" playbook uses the checked-out branch (HEAD) of each docs repo. Thus, you can make changes to the docs and see them reflected in the streaming docs site without having to commit and push.
```
